io.ts

```diff
@@ -187,12 +188,29 @@ export function muxExecutor(
   function dequeue(): void {
     const [e] = queue;
     if (!e) return;
-    sendCommand(w, r, e.command, ...e.args)
+    sendCommand(
+      connection.writer as BufWriter,
+      connection.reader as BufReader,
+      e.command,
+      ...e.args
+    )
       .then((v) => {
-        // console.log(e.command, e.args, v);
         e.d.resolve(v);
       })
-      .catch((err) => e.d.reject(err))
+      .catch(async (err) => {
+        if (
+          (
```

Comment:
`BadResource` error is thrown when using an closed connection.
So I think it' s necessary to check if the connection has been explicitly closed by the user.

```suggestion
          (
            (err instanceof Deno.errors.BadResource && !connection.isClosed) // If `connection.isClosed` is `true`, then we can assume that the connection is explicitly closed by user.
            || err instanceof Deno.errors.BrokenPipe
            || err instanceof Deno.errors.ConnectionAborted
            || err instanceof Deno.errors.ConnectionRefused
            || err instanceof Deno.errors.ConnectionReset
          )
          && attemptReconnect
```